src/isomorphic/hooks/ReactComponentTreeHook.js

```diff
@@ -92,12 +99,47 @@ function create(id, element, parentID) {
     isMounted: false,
     updateCount: 0,
   };
-  if (canUseMap) {
+
+  if (canUseCollections) {
     itemMap.set(id, item);
-    return;
+  } else {
+    var key = getKeyFromID(id);
+    itemByKey[key] = item;
+  }
+}
+
+function addRoot(id) {
+  if (canUseCollections) {
+    rootIDSet.add(id);
+  } else {
+    var key = getKeyFromID(id);
+    rootByKey[key] = true;
+  }
+}
+
+function removeRoot(id) {
+  if (canUseCollections) {
+    rootIDSet.delete(id);
+  } else {
+    var key = getKeyFromID(id);
+    delete rootByKey[key];
+  }
+}
+
+function getRegisteredIDs() {
+  if (canUseCollections) {
+    return Array.from(itemMap.keys());
+  } else {
+    return Object.keys(itemByKey).map(getIDFromKey);
+  }
+}
+
+function getRootIDs() {
+  if (canUseCollections) {
+    return Array.from(rootIDSet.keys());
+  } else {
+    return Object.keys(rootByKey).map(getIDFromKey);
```

Comment:
If we stored the id as the value then this could be Object.values(rootByKey). I suppose we don't know that's polyfilled though.